plugins/modules/nmcli.py

```diff
@@ -2105,6 +2105,12 @@ def _compare_conn_params(self, conn_info, options):
 
             if key in conn_info:
                 current_value = conn_info[key]
+                if key in ('wifi.wake-on-wlan',
+                           '802-11-wireless.wake-on-wlan') and \
+                        current_value is not None:
+                    match = re.match('0x([0-9A-Fa-f]+)', current_value)
+                    if match:
+                        current_value = str(int(match.group(1), 16))
```

Comment:
It is sufficient to check against `802-11-wireless.wake-on-wlan` which is the current value in _nmcli_

```
if key == '802-11-wireless.wake-on-wlan' and current_value is not None:
    match = re.match('0x([0-9A-Fa-f]+)', current_value)
    if match:
    current_value = str(int(match.group(1), 16))
```

Reply:
Should I not be concerned in this code with backward compatibility with older versions of nmcli that use `wifi` instead of `802-11-wireless`? Or are you saying that this code was always wrong, i.e., nmcli never used `802-11-wireless` here?

Reply:
Hm...this looks like the old `wifi` option in `nmcli` is not supported. Looking at old Ubuntu manpages make me guess the `802-11-wireless` was introduced in _NetworkManager 1.x_. 

https://manpages.ubuntu.com/manpages/trusty/man1/nmcli.1.html
https://manpages.ubuntu.com/manpages/trusty/man1/nmcli.1.html

Older versions of `nmcli` also seem to have not all commands supported that are needed, 
So I guess just using `802-11-wireless` should be fine?

Again, maybe someone with more experience on this module and NetworkManager should review this.